src/vm/comdelegate.cpp

```diff
@@ -2593,15 +2593,6 @@ BOOL COMDelegate::NeedsWrapperDelegate(MethodDesc* pTargetMD)
 {
     LIMITED_METHOD_CONTRACT;
 
-#ifdef _TARGET_ARM_
```

Comment:
I believe that implementing `COMDelegate::GetSecureInvoke` is easier. It should be implemented similar to how `COMDelegate::GetMulticastInvoke` is implemented, except that there will be no loop. It should just re-push the arguments and call the actual target. The call to actual target generated by the JIT has to pass the indirection cell in R4. There will need be a new flag passed to the JIT to make it happen.